zipline/finance/trading.py

```diff
@@ -204,17 +204,18 @@ def normalize_date(self, test_date):
         return pd.tseries.tools.normalize_date(test_date)
 
     def utc_dt_in_exchange(self, dt):
-        return pd.Timestamp(dt).tz_convert(self.exchange_tz)
+        return pd.Timestamp(dt, tz='UTC').tz_convert(self.exchange_tz)
 
     def exchange_dt_in_utc(self, dt):
         return pd.Timestamp(dt, tz=self.exchange_tz).tz_convert('UTC')
 
     def is_market_hours(self, test_date):
-        if not self.is_trading_day(test_date):
+        day = pd.Timestamp(self.utc_dt_in_exchange(test_date).date(), tz='UTC')
```

Comment:
so basically what I need is the current day in the local time and what its using the is current day in UTC but the reason my solution is a bit hacked is that sometime the function receives market open and sometimes with day time 00:00 in UTC (and perhaps also market close). Otherwise if you were always receiving the same variable it would be easier to construct a neater solution. All of the time/day functions are in desperate need of standardization.